pkg/tagger/tagstore.go

```diff
@@ -55,31 +59,41 @@ func (s *tagStore) processTagInfo(info *collectors.TagInfo) error {
 	}
 
 	// TODO: check if real change
-	s.storeMutex.RLock()
+	s.storeMutex.Lock()
+	defer s.storeMutex.Unlock()
 	storedTags, exist := s.store[info.Entity]
-	s.storeMutex.RUnlock()
-	if exist == false {
+	if !exist {
 		storedTags = &entityTags{
 			lowCardTags:  make(map[string][]string),
 			highCardTags: make(map[string][]string),
 		}
+		s.store[info.Entity] = storedTags
 	}
 
 	storedTags.Lock()
+	defer storedTags.Unlock()
 	storedTags.lowCardTags[info.Source] = info.LowCardTags
 	storedTags.highCardTags[info.Source] = info.HighCardTags
 	storedTags.cacheValid = false
-	storedTags.Unlock()
-
-	if exist == false {
-		s.storeMutex.Lock()
-		s.store[info.Entity] = storedTags
-		s.storeMutex.Unlock()
-	}
 
 	return nil
 }
 
+func computeTagsHash(tags []string) string {
+	hash := ""
+	// do not sort original slice
```

Comment:
If we do not sort, we might end up rescheduling checks from pods with several services. If the DCA returns `kube_svc:svc1` and `kube_svc:svc2` but they don't show up in this order from one run to another, it will look as though we are outdated but we aren't

Reply:
Actually, I think this might happen if the DCA returns kube deploy and kube svc (i.e. at least 2 tags) I think.